consensus/module.go

```diff
@@ -194,13 +193,24 @@ func (m *consensusModule) Start() error {
 		return err
 	}
 
-	go m.metadataSyncLoop()
-	go m.blockApplicationLoop()
+	m.wg.Add(1)
+	go func() {
+		defer m.wg.Done()
+		m.blockApplicationLoop()
+	}()
```

Comment:
I don't fully understand why we have a wait group here at all and don't just do 

`go m.blockApplicationLoop()`

and inside of the loop, we'd do something like

```
	for {
		select {
		case blockResponse := <-blocksResponsesReceived:
```